src/score_gen/score.rs

```diff
@@ -405,6 +415,12 @@ mod tests {
     use merlin::Transcript;
     use rand_core::RngCore;
 
+    #[test]
+    fn log_2_rounded() {
+        let two_pow_128 = Scalar::from(2u64).pow(&[128u64, 0, 0, 0]);
+        assert_eq!(next_pow_2_bitnum(two_pow_128), 129u64);
```

Comment:
And this is assertion is also not correct – it's assert that `next_pow_2_bitnum` returns what the implementation returns, but the logic of the implementation is wrong (bitnum + 1, plus doesn't take in account if the Scalar passed is `1` or `0`), and therefore the test.